channel/channel.go

```diff
@@ -0,0 +1,57 @@
+package channel
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+)
+
+type CryChan[T any] struct {
+	c chan T
+	t time.Duration
+}
+
+// NewCryChan creates a new cryChan of the given type and size
+func NewCryChan[T any](bufSize int, timeout time.Duration) CryChan[T] {
+	Logger = Logger.With().Int("size", bufSize).Logger()
+
+	return CryChan[T]{
+		c: make(chan T, bufSize),
+		t: timeout,
+	}
+}
+
+// Push adds an element in the channel, or logs if it's blocked for too long
+func (c *CryChan[T]) Push(e T) {
```

Comment:
There's a bit of API inconsistency. `Push` is essentially blocking, with an arbitrary timeout defined as part of the channel creation, whereas `Pop` uses a `context` and assumes the timeout is provided by the context (as needed).

Would it make sense to have separate sets of APIs ? 
`PushWithContext`
`PushWithTimeout`
`Push`
+
`PopWithContext`
`PopWithTimeout`
`Pop`
+
`Channel` (to return the raw channel)

They could all use the same underlying implementation (i.e. they could all rely on the context implementation), but it would a reasonable amount of flexibility to make using this library practical.